src/librustc/ty/maps.rs

```diff
@@ -580,10 +598,17 @@ macro_rules! define_maps {
                        key,
                        span);
 
+                profq_msg!(tcx,
+                    ProfileQueriesMsg::QueryBegin(span.clone(),
+                                                  QueryMsg::$name(format!("{:?}", key))));
```

Comment:
I wonder if we want multiple options. There are various things we might want data for: 

- Which passes take the most time? 
    - For this, we don't care about cache hits, and we probably don't care about cache *keys* either.
- Which *bits of the input* took the most time?
    - For this, we don't care about cache hits, but we **do** care about cache keys, but *maybe* only those that took significant time.
- An overview of what happened, e.g. because something went wrong with incremental (re-use failed to occur, or re-use occurred improperly) and we'd like to know why
    - For this, knowing the precise cache keys is very important, and we do care I think about cache hits.

I think this PR started off targeting the last one, and kind of grew to target the others. I suspect we care about all three. We may want to allow for options choosing which one you want, I'm not sure.